### DISCUSSION TITLE
How to add a tooltip text to a button?

### DISCUSSION THREAD

**Discussion by bigfooted** [Kitware/trame#227] [Q&A] [answered]
Hi,

I placed this question in paraview discourse but I think this it the better place to ask :
suppose I have a simple checkbox in trame. How can I add a tooltip text when I hover over the box? I do not know how I can translate the explanation on the vuetify website into trame functionality.

The vuetify site gives for instance this example:

```

<v-tooltip text="Tooltip">
  <template v-slot:activator="{ props }">
    <v-btn v-bind="props">Tooltip</v-btn>
  </template>
</v-tooltip>

```

and I have a simple checkbox like this:
```

    vuetify.VCheckbox(
        v_model=("list_change", True),
        on_icon="mdi-account-check",
        off_icon="mdi-account-check-outline",
        classes="mx-1",
        hide_details=True,
        dense=True,
        click=reset_resolution,
    )

```
I have to add a v_tooltip and a v_bind?

**Comment by jourdain** [comment, marked as answer]
```python
with vuetify.VTooltip(text="Tooltip") as elem_to_test:
    with vuetify.Template(v_slot_activator="{ props }"):
        vuetify.VCheckbox(
          v_bind="props"
          v_model=("list_change", True),
          on_icon="mdi-account-check",
          off_icon="mdi-account-check-outline",
          classes="mx-1",
          hide_details=True,
          dense=True,
          click=reset_resolution,
      )
   print(elem_to_test.html) # <--- That way you can see if it match what the vuetify doc is telling you
```

**Reply by jourdain** [reply]
Also are you sure you are looking at vuetify 2 and not 3?

You should look here https://v2.vuetifyjs.com/en/components/tooltips/#usage

Which should be 

```
<v-tooltip bottom>
      <template v-slot:activator="{ on, attrs }">
        <v-checkbox
          v-bind="attrs"
          v-on="on"
          ....
        />
      </template>
      <span>Tooltip</span>
</v-tooltip>
```

**Reply by bigfooted** [reply]
Thanks, @jourdain, I guess I was also looking at the wrong page. So the rough translation is that `<v-xyz>` usually becomes `vuetify.VXyz`, but something like `v_slot_activator="{ props }"` was a mystery to me.

**Reply by jourdain** [reply]
This is capture on the reference PDF (7 slides) of the [free course](https://kitware.github.io/trame/docs/course_intro.html).

**Reply by bigfooted** [reply]
Whyn I try your code above, the tooltip still does not appear. the print output is below, it looks like the actual text "Tooltip" is missing:

```
<v-tooltip >
<template v-slot:activator="{ attrs }">
<v-checkbox class="mx-1" v-bind="attrs" v-model="export_config_file" dense hide-details off-icon="mdi-download-box" on-icon="mdi-download-box-outline" />
</template>
</v-tooltip>
```


My code is like what you wrote:
```
  with vuetify.VTooltip(text="Tooltip text") as elem_to_test:
            with vuetify.Template(v_slot_activator="{ attrs }"):
                vuetify.VCheckbox(
                  v_bind="attrs",
                  v_model=("export_config_file", True),
                  on_icon="mdi-download-box-outline",
                  off_icon="mdi-download-box",
                  classes="mx-1",
                  hide_details=True,
                  dense=True,
                  #click=reset_resolution,
              )
            print(elem_to_test.html)
```

**Reply by jourdain** [reply]
VTooltip does not have a __text__ attribute. It needs to be provided as a template sibling.  (`<span>Tooltip</span>`)

**Reply by jourdain** [reply]
The difference is between vuetify 3 vs 2.

**Reply by bigfooted** [reply]
OK, I see. So the example for the button in the reference pdf works, that is this one:

```
       with vuetify.VTooltip(bottom=True) as debug:
            with vuetify.Template(v_slot_activator="{ on, attrs }"):
                vuetify.VBtn(
                  "Button",
                  v_bind="attrs",
                  v_on="on"
              )
            html.Span("Tooltip text")
```

but a similar example for a checkbox does not work:
 ```
       with vuetify.VTooltip(top=True) as elem_to_test:
            with vuetify.Template(v_slot_activator="{ on, attrs }"):
                vuetify.VCheckbox(
                  v_bind="attrs",
                  v_on="on",
              )
            html.Span("Tooltip text")
```

So does it not work the same for different elements (button, checkbox, sliders)?

**Reply by jourdain** [reply]
It should but maybe the Checkbox do something funky. To solve that, just nest it inside a Div.

```
      with vuetify.VTooltip(top=True) as elem_to_test:
           with vuetify.Template(v_slot_activator="{ on, attrs }"):
               with html.Div(v_bind="attrs", v_on="on"):
                   vuetify.VCheckbox(...)
           html.Span("Tooltip text")
```

**Reply by bigfooted** [reply]
Thanks, @jourdain using a Div worked, although the tooltip is now much farther away than for the button.

**Reply by jourdain** [reply]
Then that can be fixed by css/style/class

**Reply by jourdain** [reply]
Try `vuetify.VCheckbox(classes="ma-0 pa-0", ...)`

**Reply by bigfooted** [reply]
Thanks again!

**Reply by bigfooted** [reply]
For the movement of the tooltip, one should use nudge_top and nudge_bottom it seems, I had to use nudge_left=100 to get it more or less right:
`with vuetify.VTooltip(top=True, nudge_left="100",nudge_top="0") `
